src/software/sensor_fusion/filter/ball_filter.cpp

```diff
@@ -245,8 +245,17 @@ LinearRegressionResults BallFilter::getLinearRegressionLine(
         A.bdcSvd(Eigen::ComputeThinU | Eigen::ComputeThinV).solve(b);
     // How to calculate the error is from
     // https://eigen.tuxfamily.org/dox/group__TutorialLinearAlgebra.html
-    double regression_error =
-        (A * regression_vector - b).norm() / (b.norm() + 1.0e-9);  // norm() is L2 norm
+    double regression_error;
+
+    if (b.norm() != 0)
+    {
+        regression_error =
+            (A * regression_vector - b).norm() / (b.norm());  // norm() is L2 norm
+    }
+    else
+    {
+        regression_error = 0;
```

Comment:
why `regression_error = 0;`?

Reply:
It used to be max double which seemed ok

Reply:
What I was finding was that when `b.norm()` was 0, the `regression_error` would also be 0.
How `regression_error` is used is in `estimateBallState`. 
```
Line regression_line =
        x_vs_y_regression.regression_error < y_vs_x_regression.regression_error
            ? x_vs_y_regression.regression_line
            : y_vs_x_regression.regression_line;
```
What was happening before was the nominator `(A * regression_vector - b).norm() / (b.norm() + 1.0e-9)` was 0 anyways,, which caused `linear_regression` to be 0 as long as the denominator was not 0 as well. So, setting `regression_error` to `DBL_MAX` caused tests to fail. I made this change assuming if `b.norm()` were 0, then the nominator would also be 0 as that was what the tests indicated.

Reply:
I'd be more comfortable if we check the numerator ==0 too

Reply:
Done.